compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1941,6 +1942,12 @@ public void visit(BLangSimpleVariable varNode) {
         if (isDeprecated(varNode.annAttachments)) {
             varSymbol.flags |= Flags.DEPRECATED;
         }
+
+        // Skip setting the state if there's a diagnostic already (e.g., redeclared symbol)
+        if (varSymbol.type == symTable.semanticError && varSymbol.state == DiagnosticState.VALID) {
+            varSymbol.state = DiagnosticState.UNKNOWN_TYPE;
```

Comment:
Don't we have to cover optional field variables in a record?

Reply:
Cover optional fields as in?

Reply:
```
type Rec1 {
    name?;
}
```

Reply:
A scenario like this ?

Reply:
For the following case a symbol is not returned since it gets recovered as `name <missing_identifier>`
```ballerina
type Rec1 record{
    name?;
}
```

Reply:
Ah okay